benchmark-scripts/src/functions.py

```diff
@@ -180,11 +191,71 @@ def remove_weaviate_class(client):
         remove_weaviate_class(client)
 
 
-def import_into_weaviate(client, efConstruction, maxConnections, benchmark_file):
+def import_data_slice_to_weaviate(
+        weaviate_url: str,
+        batch_size: int,
+        vectors: Sequence,
+        subprocess_number: int,
+        data_start_index: int,
+    ):
+    """
+    Import a slice of the dataset in a different process (core).
+    On exceptions during import saves the global counter to a sub-process specific
+    file (to avoid writing to the same file at once).
+    """
+    
+    try:
+        client = Client(weaviate_url, timeout_config=(5, 60))
+        client.batch.configure(
+            timeout_retries=10,
+        )
+    except Exception as error:
+        loguru.logger.exception(
+            f"sub-process {subprocess_number}: Can't connect to Weaviate, is it running?"
+        )
+        raise error
+    
+    counter = 0
+    batch_c = 0
+    vector_len = len(vectors)
+    loguru.logger.info(
+        f'Start import sub-process {subprocess_number}, vectors start index {data_start_index}'
+    )
+    try:
+        for vector in vectors:
+            client.batch.add_data_object(
+                data_object={'counter': counter + data_start_index},
+                class_name='Benchmark',
+                uuid=str(uuid.uuid3(uuid.NAMESPACE_DNS, str(counter + data_start_index))),
+                vector = vector
+            )
+            if batch_c == batch_size:
+                add_batch(client, counter, vector_len, subprocess_number)
+                batch_c = 0
+            counter += 1
```

Comment:
Are the values for `counter` still globally unique? I think they are used in the valuation part to check for recall. Maybe @bobvanluijt can confirm. I.e. object number 1,234,567 of the dataset would need to have this field set to excatly 1,234,567. Is this still the case with the slice importing?

Reply:
Oh, I think I found the answer a few lines up where it is used as a sum with `data_start_index`, but maybe you can still confirm that I'm not missing anything here. Thanks.